### ISSUE TITLE
Implement serialization for tasks

### ISSUE THREAD

**Issue by collinarnett** [collinarnett/PersonalGamification#23]
Tasks must be serialized in order to have a persistent state. There are examples of how to serialize in previous commits to the dev branch. All reading from directories should be done at the beginning of the program run and all writing should be done at the end of the program run. An object `Writer` should be created to hold serialization methods and the writer should be able to process classes other than just tasks.

**Comment by abene86** [comment]
I used the os-lib library to handle file Io operation. My rationale behind this decision is pretty easy to create dirs, files and delete, move them, and more without worrying about the low level of Java file operation. 

I created a writer class and wrote two functions createneededdirectory and searlize. The first function takes in a file path where we want to create the folder and the folder name. It then checks if the folder has already been created and if it has it does nothing. If the folder does not exist, it creates it. For serialization, it takes in filepath, filename, and the object we want to write to the file, and it then uses yaml mapper to serialize it. I wrote a simple test by creating a personal object and serializing it. 

I tested whether it would work to serialize in pwd and in a separate folder.

The theTaskInProgress folder was created as a result of testing the createdNeededDir function.
